server/src/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1078,6 +1082,16 @@ public void cleanupStorage(boolean recurring) {
                         }
                     }
 
+                    //destroy snapshots in destroying state in snapshot_store_ref
+                    List<SnapshotDataStoreVO>  ssSnapshots = _snapshotStoreDao.listByState(ObjectInDataStoreStateMachine.State.Destroying);
+                    for(SnapshotDataStoreVO ssSnapshotVO : ssSnapshots){
+                        try {
+                            _snapshotService.deleteSnapshot(snapshotFactory.getSnapshot(ssSnapshotVO.getSnapshotId(), DataStoreRole.Image));
```

Comment:
deleteSnapshot() is going to send an agent command to cleanup the snapshot from secondary, if there is any failure will cleanup be retried or the DB entry is simply updated?